src/plugins/discover/public/application/components/doc_viewer/doc_viewer_tab.tsx

```diff
@@ -53,17 +53,11 @@ export class DocViewerTab extends React.Component<Props, State> {
   }
 
   render() {
-    const { component, render, renderProps, title } = this.props;
+    const { component: Component, render, renderProps, title } = this.props;
     const { hasError, error } = this.state;
 
     if (hasError && error) {
       return <DocViewerError error={error} />;
-    } else if (!render && !component) {
```

Comment:
:information_source: I actually reordered that if's (though the logic stays the same), but that way TypeScript is able to detect down properly, that either `render` OR `Component` must be set. It's unfortunately not smart enough to detect that from this `if` and the `if(render)` that after that `Component` can't be `undefined` anymore.